src/python/WMCore/WMRuntime/Scripts/SetupCMSSWPset.py

```diff
@@ -680,15 +686,17 @@ def __call__(self):
 
         try:
             origCores = int(getattr(self.step.data.application.multicore, 'numberOfCores', 1))
+            eventStreams = int(getattr(self.step.data.application.multicore, 'eventStreams', 0))
             resources = {'cores': origCores}
-            resizeResources(resources)
+            if not eventStreams:  # Don't know how to handle resizing jobs with different nThreads and nStreams
```

Comment:
Either is going to be pathological. We have to make sure it never happens. If we fire up too many cores, yeah, we do something bad and we run out of memory. If we scale down to four course, it probably still runs out of memory since we were setting nStreams < nCores for a reason.

So maybe your suggestion is "less bad" but like I said, we have to make sure this does not happen.